Src/ReadCombustionTankPressure.c

```diff
@@ -0,0 +1,85 @@
+#include "stm32f4xx.h"
+#include "stm32f4xx_hal_conf.h"
+#include "cmsis_os.h"
+#include "math.h"
+
+#include "ReadCombustionTankPressure.h"
+
+#include "Data.h"
+
+#define ADC1_QUEUE_SIZE 5
+
+static int READ_COMBUSTION_TANK_PRESSURE_PERIOD = 300;
+
+static const int ADC1_POLL_TIMEOUT = 150;
+
+static uint16_t adc1ValuesQueue[ADC1_QUEUE_SIZE] = {0};    // Average ADC1_QUEUE_SIZE ADC readings that are each 12 bits in size
+// ( maximum resolution for this ADC )
+
+void readCombustionTankPressureTask(void const* arg)
+{
+    CombustionTankPressureData* data = (CombustionTankPressureData*) arg;
+    uint32_t prevWakeTime = osKernelSysTick();
+
+    double vo = 0;  // The voltage across the 133k resistor
+    double vi = 0;  // The pressure sensor output
+    double tankPressure = 0;
+
+    int adc1QueueIndex = 0;    // Counts up to ADC1_QUEUE_SIZE ADC readings before averaging
+
+    // Resistor values in kOhms
+    int R1 = 100;
+    int R2 = 133;
+
+    HAL_ADC_Start(&hadc1);  // Start the ADC peripheral
+
+    osDelay(5);
+
+    for (;;)
+    {
+        osDelayUntil(&prevWakeTime, READ_COMBUSTION_TANK_PRESSURE_PERIOD);
+
+        if (HAL_ADC_PollForConversion(&hadc1, ADC1_POLL_TIMEOUT ) == HAL_OK)
+        {
+            adc1ValuesQueue[adc1QueueIndex++] = HAL_ADC_GetValue(&hadc1);
+        }
+
+        if (adc1QueueIndex >= ADC1_QUEUE_SIZE)
```

Comment:
The purpose of the running average is so that we don't need this. We can always take an average.

Example, the array might look like this per iteration

```
56, 0, 0, 0, 0
56, 59, 0, 0, 0
56, 59, 52, 0 ,0
56, 59, 52, 55, 0
56, 59, 52, 55, 60
54, 59, 52, 55, 60
54, 55, 52, 55, 60
54, 55, 57, 55, 60
54, 55, 57, 50, 60
```

So the first 4 readings won't be accurate because the array hasn't been fully filled yet, but past that. Every average of the whole array is a good reading. As it stands right now, with a period of 300 and only updating the tank pressure data every 5 reads, we're updating the tank pressure data every 1.5 seconds. That's very slow.